File: backend/OctoFlashcardStudy.API/Domain/Entities/FlashCard.cs
namespace OctoFlashcardStudy.API.Domain.Entities
{
    public class FlashCard
    {
        public Guid Id { get; set; }
        public Guid DeckId { get; set; }

        public string Term { get; set; } = string.Empty;
        public string TermLanguage { get; set; } = string.Empty;
        public string Definition { get; set; } = string.Empty;
        public string DefinitionLanguage { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public Deck Deck { get; set; } = null!;
    }
}

File: frontend/src/types/flashCard.ts
export interface CreateFlashCardRequest{
    term: string;
    termLanguage: string;
    definition: string;
    definitionLanguage: string;
}

export interface UpdateFlashCardRequest {
    term: string;
    termLanguage: string;
    definition: string;
    definitionLanguage: string;
}

export interface FlashCardResponse {
    id: string;
    deckId: string;
    term: string;
    termLanguage: string;
    definition: string;
    definitionLanguage: string
    createAt: string;
    updateAt: string;
}